src/core/thread/topology.cpp

```diff
@@ -274,6 +278,87 @@ void Child::GenerateChallenge(void)
     IgnoreError(Random::Crypto::FillBuffer(mAttachChallenge, sizeof(mAttachChallenge)));
 }
 
+#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE
+bool Child::HasMlrRegisteredAddress(const Ip6::Address &aAddress)
+{
+    bool         has = false;
+    Ip6::Address address;
+
+    VerifyOrExit(mMlrRegisteredMask.HasAny(), OT_NOOP);
+
+    for (Child::Ip6AddressIterator iter; GetNextIp6Address(iter, address) == OT_ERROR_NONE;)
+    {
+        if (mMlrRegisteredMask.Get(iter.Get()) && address == aAddress)
+        {
+            ExitNow(has = true);
+        }
+    }
+
+exit:
+    return has;
+}
+
+MlrState Child::GetAddressMlrState(uint16_t aAddressIndex)
+{
+    bool     toRegister = mMlrToRegisterMask.Get(aAddressIndex);
+    bool     registered = mMlrRegisteredMask.Get(aAddressIndex);
+    MlrState mlrState;
+
+    OT_ASSERT(!(toRegister && registered));
+
+    if (toRegister)
+    {
+        mlrState = kMlrStateToRegister;
+    }
+    else if (registered)
+    {
+        mlrState = kMlrStateRegistered;
+    }
+    else
+    {
+        mlrState = kMlrStateRegistering;
+    }
+
+    return mlrState;
+}
+
+void Child::SetAddressMlrState(uint16_t aAddressIndex, MlrState aState)
+{
+    bool toRegister = false;
+    bool registered = false;
+
+    switch (aState)
+    {
+    case kMlrStateRegistering:
+        break;
+    case kMlrStateToRegister:
+        toRegister = true;
+        break;
+    case kMlrStateRegistered:
+        registered = true;
+        break;
+    }
+
+    if (toRegister)
+    {
+        mMlrToRegisterMask.Set(aAddressIndex);
```

Comment:
How about adding `Set(index, bool)` in `BitVector` to set a bit at a given index to a given value?
(maybe in future PR?).

Reply:
I also felt the need to have this method. Let's do it now!